client/index.js

```diff
@@ -21,22 +22,37 @@ function App() {
 
   function getTopMovies(){
     let movieData = [];
-    getTopMoviesApi().then(data => {
-        data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
-        setMovies(movieData);
-        localStorage.setItem('topMovies', movieData);
-    })
-    .catch(err => console.log(err));
+    console.log("getting top movies");
+    let cachedMovies = localStorage.getItem('topMovies');
+    if (cachedMovies){
+        setMovies(JSON.parse(cachedMovies));
+    } else {
+      getTopMoviesApi().then(data => {
+          // console.log(data);
+          console.log("getting top movies");
+          data.results.forEach((x, i) => movieData.push({"Movie": x.title, "Overview": x.overview, "Release Date": x.release_date, "Language": x.original_language}));
+          // console.log("top movies", movieData);
+          setMovies(movieData);
+          localStorage.setItem('topMovies', JSON.stringify(movieData));
+      })
+      .catch(err => console.log(err));
+  }
+    // console.log("top movies", data);
   }
 
   function searchMovies(){
     let movieData = [];
+    let cachedMovies = localStorage.getItem(searchQuery);
```

Comment:
much better here

- API still does API things
- decide whether we want to hit API in the calling method

👍 👍